lib/autocomplete-manager.coffee

```diff
@@ -272,18 +186,166 @@ class AutocompleteManager
     @editor.insertText(match.word)
     @editor.setSelectedBufferRanges(newSelectedBufferRanges)
 
-  # Private: Sets the current buffer, starts listening to change events and delegates
-  # them to #onChanged()
+  # Public: Finds and returns the content before the current cursor position
+  #
+  # selection - The {Selection} for the current cursor position
+  #
+  # Returns {String} with the prefix of the {Selection}
+  prefixOfSelection: (selection) ->
+    selectionRange = selection.getBufferRange()
+    lineRange = [[selectionRange.start.row, 0], [selectionRange.end.row, @editor.lineTextForBufferRow(selectionRange.end.row).length]]
+    prefix = ''
+    wordRegex = /\b\w*[a-zA-Z_-]+\w*\b/g
```

Comment:
We should really make this select the prefix by the innermost scope, that way we utilize the language syntax parsing to hint at what is really part of what we are replacing. Might need to make this configurable also so that languages that have bad syntax detection don't break.

Not needed for this round, but we should think about it.